### PR TITLE
Run instrumentation tests on CI

### PR DESCRIPTION
**Pull request by ellykits**
Signed-off-by: Elly Kitoto <junkmailstoelly@gmail.com>

**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] 

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by pld** [comment]
@ndegwamartin is something wrong with the codecov calculations?

**Comment by ndegwamartin** [comment]
> @ndegwamartin is something wrong with the codecov calculations?

The calculations are fine, as communicated earlier we did expected some coverage drop because:

- We refactored UI components to use Compose with are actual Kotlin classes (vs XML files for UI)
- For the PR the Quest, coverage was previously un-reported meaning after the fix coverage would drop **(Engine + Quest)/2**

**Comment by ndegwamartin** [comment]
The next steps for this PR would be: 

- Merging in the PR
- Commencing the writing of instrumentation tests cc @pld @Mstjamush @ellykits

**Comment by pld** [comment]
K sounds good, we should dedicated to improving the coverage with more unit tests too, I'll look through the pr

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,7 +16,10 @@ env:
 
 jobs:
   engine-tests:
-    runs-on: ubuntu-latest
+    runs-on: macos-latest
```
Comment (pld) [inline .github/workflows/ci.yml:19]: intended?

File: android/engine/src/test/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistryTest.kt [hunk 2]
```diff
@@ -211,6 +212,7 @@ class ConfigurationRegistryTest : RobolectricTest() {
   }
 
   @Test
+  @Ignore("Fix this flaky test")
```
Comment (pld) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistryTest.kt:215]: I added an issue for this, https://github.com/opensrp/fhircore/issues/1423

File: android/quest/build.gradle [hunk 3]
```diff
@@ -34,7 +34,8 @@ android {
         buildConfigField("String", 'OAUTH_CLIENT_SECRET', "\"${OAUTH_CLIENT_SECRET}\"")
         buildConfigField("String", 'OAUTH_SCOPE', "\"${OAUTH_SCOPE}\"")
 
-        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+  //      testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
```
Comment (pld) [inline android/quest/build.gradle:37]: can we remove this line?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ndegwamartin) [reply .github/workflows/ci.yml:19]: Yeah, this is required in order to run the device emulator on the cloud (requires hardware acceleration)

**Review by pld — APPROVED** [review]

File: android/engine/src/test/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistryTest.kt (hunk 2 above)
Comment (ndegwamartin) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistryTest.kt:215]: Noted

File: android/quest/build.gradle (hunk 3 above)
Comment (ndegwamartin) [reply android/quest/build.gradle]: Done

**Comment by FikriMilano** [comment]
restoring the branch to investigate why androidTest is not reporting code coverage at the moment

**Pull request merged**